mllib/src/main/scala/org/apache/spark/mllib/feature/Word2Vec.scala

```diff
@@ -281,16 +280,17 @@ class Word2Vec extends Serializable with Logging {
     val expTable = sc.broadcast(createExpTable())
     val bcVocab = sc.broadcast(vocab)
     val bcVocabHash = sc.broadcast(vocabHash)
-
-    val sentences: RDD[Array[Int]] = words.mapPartitions { iter =>
+    //each partition is a collection of sentences, will be translated into arrays of Index integer
+    val sentences: RDD[Array[Int]] = dataset.mapPartitions { iter =>
```

Comment:
I suspect the max length here is mostly because it was in the original implementation. The problem here is: what if the 'sentence' ends up being very very long?

Although a real "sentence" would not ever reasonably be 1000+ words, word2vec is used in other contexts where a "sentence" is something else. I can see making this configurable, but not removing it.